numpy/core/_add_newdocs.py

```diff
@@ -7969,62 +7969,199 @@ def luf(lamdaexpr, *args, **kwargs):
+    """
+    Signed integer type, compatible with C ``unsigned long long``.
+    Character code: ``'Q'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'ulonglong', unsigned_integer_aliases)))
+
+add_newdoc('numpy.core.numerictypes', 'half',
+    """
+    Half-precision floating-point number type.
+    Character code ``'e'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'half', float_aliases)))
 
-add_newdoc('numpy.core.numerictypes', 'float128',
+add_newdoc('numpy.core.numerictypes', 'single',
     """
-    128-bit floating-point number. Character code: 'g'. C long float
-    compatible.
+    Single-precision floating-point number type, compatible with C ``float``.
+    Character code ``'f'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'single', float_aliases)))
 
+add_newdoc('numpy.core.numerictypes', 'float_',
+    """
+    Double-precision floating-point number type, compatible with Python `float`
+    and C ``double``.
+    Character code ``'d'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'float_', float_aliases)))
+
+add_newdoc('numpy.core.numerictypes', 'longfloat',
+    """
+    Extended-precision floating-point number type, compatible with C
+    ``long double`` but not necessarily with IEEE 754 quadruple-precision. 
+    Character code: ``'g'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'longfloat', float_aliases)))
+
+add_newdoc('numpy.core.numerictypes', 'float96',
+    """
+    96-bit extended-precision floating-point number type.
+    
     """)
 
-add_newdoc('numpy.core.numerictypes', 'int8',
-    """8-bit integer. Character code ``b``. C char compatible.""")
 
-add_newdoc('numpy.core.numerictypes', 'int16',
-    """16-bit integer. Character code ``h``. C short compatible.""")
+add_newdoc('numpy.core.numerictypes', 'csingle',
+    """
+    Complex number type composed of two single-precision floating-point
+    numbers.
+    Character code: `'F'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'csingle', complex_aliases)))
 
-add_newdoc('numpy.core.numerictypes', 'int32',
-    """32-bit integer. Character code 'i'. C int compatible.""")
+add_newdoc('numpy.core.numerictypes', 'complex_',
+    """
+    Complex number type composed of two double-precision floating-point
+    numbers, compatible with Python `complex`.
+    Character code: ``'D'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'complex_', complex_aliases)))
 
-add_newdoc('numpy.core.numerictypes', 'int64',
-    """64-bit integer. Character code 'l'. Python int compatible.""")
+add_newdoc('numpy.core.numerictypes', 'clongfloat',
+    """
+    Complex number type composed of two extended-precision floating-point
+    numbers.
+    Character code: ``'G'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'clongfloat', complex_aliases)))
+
+add_newdoc('numpy.core.numerictypes', 'complex192',
```

Comment:
This one is an alias, not a real type

Reply:
Does that mean that `float96`is also an alias, and either `float96` or `float128` is matched against `longdouble`?